frontend/packages/shared/src/contexts/ServicesContext.tsx

```diff
@@ -39,25 +40,35 @@ const handleError = (
 ): void => {
   // TODO : log axios errors
 
-  if (error?.response?.status === ServerCodes.Unauthorized) {
-    const errorMessageKey = 'api_errors.Unauthorized';
+  const renderToast = (key: string, options: ToastOptions = {}) => {
+    const errorMessageKey = `api_errors.${key}`;
     if (i18n.exists(errorMessageKey)) {
-      toast.error(t(errorMessageKey), { toastId: errorMessageKey, autoClose: LOG_OUT_TIMER_MS });
+      toast.error(t(errorMessageKey), {
+        toastId: errorMessageKey,
+        ...options,
+      });
+    } else {
+      toast.error(t('general.error_message'), {
+        toastId: errorMessageKey,
+        ...options,
+      });
     }
+  };
+
+  const errorCode = error?.response?.data?.errorCode;
+  const unAuthorizedErrorCode = error?.response?.status === ServerCodes.Unauthorized;
+
+  if (unAuthorizedErrorCode) {
+    renderToast(errorCode || 'Unauthorized', {
+      autoClose: LOG_OUT_TIMER_MS,
+    });
     setTimeout(() => {
       logout().then(() => window.location.assign(userLogoutAfterPath()));
     }, LOG_OUT_TIMER_MS);
-    return;
   }
```

Comment:
This return is important to prevent further code execution
```suggestion
    return;
  }
```